MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/extrusions/RotationExtrusionActivity.java

```diff
@@ -0,0 +1,182 @@
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.style.functions.Function;
+import com.mapbox.mapboxsdk.style.functions.stops.IdentityStops;
+import com.mapbox.mapboxsdk.style.layers.FillExtrusionLayer;
+
+import static com.mapbox.mapboxsdk.style.layers.Filter.eq;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillExtrusionBase;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillExtrusionColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillExtrusionHeight;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillExtrusionOpacity;
+
+/**
+ * Change the bearing and tilt of camera when viewing extrusions to get full 3D effect
+ */
+public class RotationExtrusionActivity extends AppCompatActivity implements SensorEventListener {
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private SensorManager sensorManager;
+  private Sensor gyro;
+  private Sensor magnetic;
+  private float[] gravArray;
+  private float[] magneticArray;
+  private float azimut;
+  private float pitch;
+  private float roll;
+  private float[] inclinationMatrix;
+  private float[] rotationMatrix;
+
+  //amplifiers that translate small movements in phone orientation into larger viewable map changes
+  //pitch is negative to compensate for the negative readings from the device while face up
+  //90 is used based on the viewable angle when viewing the map (from phone being flat to facing you
+  private int pitchAmplifier = -90;
+  private int bearingAmplifier = 90;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_extrusion_rotation);
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+        @Override
+        public void onMapReady(@NonNull final MapboxMap map) {
+          mapboxMap = map;
+          setupBuildings();
+        }
+    });
+
+    //initialize sensors
+    sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+
+    gyro = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
```

Comment:
We should verify sensors before using them

> Always verify that a sensor exists on a device before you attempt to acquire data from it. Don't assume that a sensor exists simply because it's a frequently-used sensor. Device manufacturers are not required to provide any particular sensors in their devices. - [Best Practices for Accessing and Using Sensors](https://developer.android.com/guide/topics/sensors/sensors_overview.html#sensors-practices)